openeuler/openeuler.go

```diff
@@ -0,0 +1,136 @@
+package openeuler
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"log"
+	"net/url"
+	"path"
+	"path/filepath"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/cheggaaa/pb"
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/vuln-list-update/utils"
+)
+
+var (
+	cvrfURL      = "https://repo.openeuler.org/security/data/cvrf"
+	retry        = 5
+	concurrency  = 20
+	wait         = 1
+	openeulerDir = "openeuler"
+)
+
+type Config struct {
+	VulnListDir string
+	URL         string
+	AppFs       afero.Fs
+	Retry       int
+}
+
+func NewConfig() Config {
+	return Config{
+		VulnListDir: utils.VulnListDir(),
+		URL:         cvrfURL,
+		AppFs:       afero.NewOsFs(),
+		Retry:       retry,
+	}
+}
+
+func (c Config) Update() error {
+	log.Print("Fetching openEuler CVRF data...")
+	u, err := url.Parse(c.URL)
+	if err != nil {
+		return xerrors.Errorf("failed to parse openEuler URL: %w", err)
+	}
+	baseURL := u.Path
+	u.Path = path.Join(baseURL, "index.txt")
+
+	res, err := utils.FetchURL(u.String(), "", c.Retry)
+	if err != nil {
+		return xerrors.Errorf("Cannot download openEuler CVRF list: %v", err)
+	}
+	cvrfUrlsMap := make(map[string][]string)
+	scanner := bufio.NewScanner(bytes.NewReader(res))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == "" {
+			continue
+		}
+		parts := strings.SplitN(line, "/", 2)
```

Comment:
Later it will be easier to understand why this is needed.
wdyt?
```suggestion
		// Take year from url
		// e.g. `2021/cvrf-openEuler-SA-2021-1001.xml` -> `2021`
		year, _, _ := strings.Cut(line, "/")
```